### PR TITLE
fix(schematics): template schematics skipping control flow blocks

### PR DESCRIPTION
**Pull request by Supamiu**
## Description

With new control flow syntax, Angular added new nodes to the `AngularCompilerLib`, which aren't `TmplAstTemplate`, meaning that the visit methods were just skipping them.

This PR adds proper handling of these node types in the `HtmlAst.visit` method so every `visitXXX` method will now just see their content.

Parsing the `BlockNode`s themselves can be added, not sure it's relevant since it'd mean that we're making a schematic that has something to do in these blocks, but adding a method to visit them is quite trivial.

-----

-----

### CODE REVIEW

**Comment by LuccaIntegration** [comment]
:woman_cook: https://lucca-front.lucca.tech/PR-2712/storybook

File: packages/ng/schematics/lib/html-ast.ts [hunk 1]
```diff
@@ -72,8 +72,21 @@ export class HtmlAstVisitor<TNode extends TemplateNode> {
...
+					this.visit(cb, branch.children);
+				});
+			} else if (node instanceof this.lib.TmplAstForLoopBlock) {
+				if (node.empty) {
+					// If we have an @empty block, visit its children too
+					this.visit(cb, [...node.children, ...node.empty.children]);
+				} else {
+					// Else, just visit @for's children
+					this.visit(cb, node.children);
+				}
```
Comment (GuillaumeNury) [inline packages/ng/schematics/lib/html-ast.ts:87]:
```suggestion
				// Visit @for's children
				this.visit(cb, node.children);

				if (node.empty) {
					// If we have an @empty block, visit its children too
					this.visit(cb, node.empty.children);
				} 
```

**Comment by LuccaIntegration** [comment]
:woman_cook: https://lucca-front.lucca.tech/PR-2712/storybook

**Review by GuillaumeNury — DISMISSED** [review]

**Comment by LuccaIntegration** [comment]
:woman_cook: https://lucca-front.lucca.tech/PR-2712/storybook

**Review by obraud — APPROVED** [review]

**Comment by LuccaIntegration** [comment]
:woman_cook: https://lucca-front.lucca.tech/PR-2712/storybook

**Review by GuillaumeNury — APPROVED** [review]

**Comment by CCNET-iLucca** [comment]
[Tests d'interfaces](https://qa.lucca.tech/15802/ui-diff)

**Pull request merged**